src/Razor/src/Microsoft.CodeAnalysis.Razor.Workspaces/ProjectSystem/GeneratedDocumentTextLoader.cs

```diff
@@ -32,6 +36,10 @@ public override async Task<TextAndVersion> LoadTextAndVersionAsync(Workspace wor
         {
             var output = await _document.GetGeneratedOutputAsync().ConfigureAwait(false);
 
+            // After generating output for our DocumentSnapshot we clear the stored state to allow for the GC to clean up any unused SyntaxTree's/IR documents.
+            // If another system needs to re-compute the generated output they can do so it just wont be cached.
+            _document.ClearStoredState();
```

Comment:
I don't love doing this in the `GeneratedDocumentTextLoader` but it's the final place where we hand off our generated document to Roslyn; meaning if I were to clear it anywhere else/sooner it'd be regenerated at this point.